### ISSUE TITLE
Inconsistent behavior between Ctrl+N and Down keybindings

### ISSUE THREAD

**Issue by akirabaruah** [nushell/nushell#13946]
### Describe the bug

`Ctrl+N` and `Down` should have identical behavior in Emacs mode, but `Ctrl+N` seems to invoke the completion menu, while `Down` simply moves down the command history (as expected).

### How to reproduce

1. Run some commands consecutively to generate history:
```
mkdir /tmp/test
cd /tmp/test
ls
touch file1
touch file2
```

2. Input the `Up` arrow key 3 times to and observe the 3 previous commands populating the prompt: `touch file2`, `touch file1`, `ls`

3. Input the `Down` key 3 times and observe the reverse of Step 2: `touch file1`, `touch file2`, `<empty prompt>`

4. Input `Ctrl+P` 3 times to view the last 3 commands and observe the same results as Step 2, ending with `ls ` in the prompt.

5. Input `Ctrl+N` 3 times and observe the completion menu open, with each key press selecting the next item in the menu.

### Expected behavior

tl;dr Step 3 (`Down` key) shows expected behavior. Step 5 (`Ctrl+N`) shows behavior that's only expected after triggering the completion menu.

Based on the [Reedline keybindings docs](https://www.nushell.sh/book/line_editor.html#default-keybindings) and the command output below, I'd expect `Ctrl+N` and `Down` to behave identically.

```nu
> keybindings default | where code =~ "(?i)down|char\\('n'\\)"                                                                                                                                                        09/27/2024 09:12:26 PM
 #     mode            modifier            code                 event
──────────────────────────────────────────────────────────────────────────────────
 0   emacs       KeyModifiers(0x0)       Down        UntilFound([MenuDown, Down])
 1   emacs       KeyModifiers(CONTROL)   Char('n')   UntilFound([MenuDown, Down])
 2   vi_normal   KeyModifiers(0x0)       Down        UntilFound([MenuDown, Down])
 3   vi_normal   KeyModifiers(CONTROL)   Char('n')   UntilFound([MenuDown, Down])
 4   vi_insert   KeyModifiers(0x0)       Down        UntilFound([MenuDown, Down])
 5   vi_insert   KeyModifiers(CONTROL)   Char('n')   UntilFound([MenuDown, Down])
```

I don't often use the arrow keys and instead rely on the Emacs style keybindings (Ctrl+P/Ctrl+N) to move up/down the command history. I've observed this expected behavior in all other shells I've used including Bash, Fish, and Zsh, so I was surprised when it didn't work this way in Nu. It's also not consistent based on the docs as far as I can tell.

### Configuration

```nu
> version | transpose key value | to md --pretty                                                                                                                                                                      09/27/2024 09:18:23 PM
| key                | value                                          |
| ------------------ | ---------------------------------------------- |
| version            | 0.98.0                                         |
| major              | 0                                              |
| minor              | 98                                             |
| patch              | 0                                              |
| branch             |                                                |
| commit_hash        |                                                |
| build_os           | macos-aarch64                                  |
| build_target       | aarch64-apple-darwin                           |
| rust_version       | rustc 1.81.0 (eeb90cda1 2024-09-04) (Homebrew) |
| cargo_version      | cargo 1.81.0                                   |
| build_time         | 2024-09-18 07:48:46 +00:00                     |
| build_rust_channel | release                                        |
| allocator          | mimalloc                                       |
| features           | default, sqlite, trash                         |
| installed_plugins  |                                                |
```

**Comment by weirdan** [comment]
Have you tried removing this part from your config?

https://github.com/nushell/nushell/blob/8200831b07ab87ab23c4a56a8a08251d0332b114/crates/nu-utils/src/sample_config/default_config.nu#L406-L418

**Comment by akirabaruah** [comment]
Yeah, I [fixed it for myself locally](https://github.com/akirabaruah/configs/commit/cc1535a353f777ba8c246f49c4dd3305b6aac400): I actually left `ide_completion_menu` there and mapped it to `Alt+Tab` instead while mapping `move_down` to `Ctrl+N`.

For this Issue, I'm thinking it'd make sense to either:

1. Update the default bindings as above (`Ctrl+N`: `move_down`, `Alt+Tab`: `ide_completion_menu`)
2. Update the docs to represent the current default mapping (`Ctrl+N`: `ide_completion_menu`, `Ctrl+T`: `move_down`)

I'm personally in favor of 1, since it better represents Emacs bindings and matches the default experience in other shells. Happy to put up the change if others agree.

More context: `ide_completion_menu` was added in https://github.com/nushell/nushell/pull/11593. Seems like `Ctrl+N` was chosen arbitrarily, even though it's actually commonly used for actions like `move_down`. @fdncred @maxomatic458 Any reason not to use `Alt+Tab` instead of `Ctrl+N` for `ide_completion_menu` in the default config?

**Comment by maxomatic458** [comment]
> Any reason not to use Alt+Tab instead of Ctrl+N for ide_completion_menu in the default config?

``Alt+Tab`` would not work great on windows (already a shortcut).

i believe the other menus are also all shortcuts with ``Ctrl`` so for consistency it would be better to have all the shortcuts (for menus) be with ``Ctrl``

**Comment by fdncred** [comment]
Come to think of it, this one should be in there too
```nushell
  {
    name: ide_completion_menu
    modifier: control
    keycode: char_p
    mode: [emacs vi_normal vi_insert]
    event: {
      until: [
        { send: menu name: ide_completion_menu }
        { send: menuprevious }
        { edit: complete }
      ]
    }
  }
```

**Comment by akirabaruah** [comment]
> `Alt+Tab` would not work great on windows (already a shortcut).
> 
> i believe the other menus are also all shortcuts with `Ctrl` so for consistency it would be better to have all the shortcuts (for menus) be with `Ctrl`

Gotcha. In that case, any reason not to use a non-conflicting binding like `Ctrl+S`?

It still seems a bit strange that `Ctrl+N` and `Ctrl+P` would be overridden as `ide_completion_menu` in the default config, given that [Reedline uses those keys for (menu) up and down](https://github.com/nushell/reedline/blob/4634f71705785a772c894aa45e30db88f3c7a8d5/src/edit_mode/keybindings.rs#L190-L200). The documentation I read in https://github.com/nushell/nushell/issues/13946#issue-2553999653 was for also Reedline, not Nushell. But as a new user, I was confused by this mismatch in associated docs and default behavior.

> Come to think of it, this one should be in there too

Yeah, that would be more consistent. https://github.com/nushell/nushell/pull/13955 addresses this.

Also, why is `Ctrl+T` bound to (menu) down [here](https://github.com/nushell/nushell/blob/8200831b07ab87ab23c4a56a8a08251d0332b114/crates/nu-utils/src/sample_config/default_config.nu#L636)? I saw it paired with `Ctrl+P` for (menu) up. For Emacs bindings, `Ctrl+T` should just transpose letters, as already defined [here](https://github.com/nushell/nushell/blob/8200831b07ab87ab23c4a56a8a08251d0332b114/crates/nu-utils/src/sample_config/default_config.nu#L768-L774), overriding this binding. Seems like unintentional dead code?

**Comment by sholderbach** [comment]
I don't know of the top of my head under which conditions the TTY flow control will interpret `Ctrl-S` as the XOFF signal that interrupts I/O processing between the terminal and the running process. With my terminal it currently gets ignored in Nushells REPL but interrupts bash (keypresses are ignored/buffered until you hit `Ctrl-Q`). If we don't control all the variables here, setting stuff to `Ctrl-S` may be a bad default.

**Comment by akirabaruah** [comment]
Proposing `Ctrl+<space>` in #13955 for now. There will likely be conflicts with any key binding we choose for this completion menu, so at some point, we should just pick one and allow users to configure according to their system requirements. Just thinking that `Ctrl+n`/`Ctrl+p` shows up as down/up much more commonly than other potential bindings conflicts, hence my aversion to using `Ctrl+n`/`Ctrl+p` for this menu.

**Comment by fdncred** [comment]
also, we have to keep in mind that if your terminal doesn't respond to `ctrl+space` it won't work. for instance, some people already have a global keybinding hook that is tied to ctrl+space. this will prevent terminals from "hearing" the shortcut so it won't be useful. i agree that we should pick something and live with it because everyone's setup is different.
